migrate/src/main/scala/migrate/ScalaMigrat.scala

```diff
@@ -9,11 +9,13 @@ import compiler.interfaces.Scala3Compiler
 import migrate.internal._
 import migrate.utils.FileUtils
 import migrate.utils.ScalaExtensions._
+import migrate.utils.ScalaMigrateReporter
 import migrate.utils.ScalafixService
 import migrate.utils.Timer._
-import scalafix.interfaces.{ ScalafixEvaluation }
+import scalafix.interfaces.ScalafixEvaluation
 
 class ScalaMigrat(scalafixSrv: ScalafixService) {
+  private val reporter = ScalaMigrateReporter
```

Comment:
Rather than creating your own `xsbti.Reporter` you should use the sbt Reporter from the task `compilerReporter`. 

Unfortunately this task is private to the `sbt` package, but you can still get it by creating a `sbt` package in your own code and then create a `val sbtCompilerReporter: TaskKey[xsbti.Reporter] = Defaults.compilerReporter` inside it.

Reply:
I don't understand fully. If the reporter is the one used from sbt and then passed through the plugin, which one I will use in my tests?

Reply:
Maybe you can mock one.

But I I am no even you need a reporter at all. See comments below: